modules/embedding-microshift-into-rpm-ostree-image.adoc

```diff
@@ -0,0 +1,212 @@
+
+=== Adding the MicroShift container images
+Image Builder supports building rpm-ostree images with embedded container
+images. Embedded container images are immediately available to container
+engines like podman and cri-o after booting, without having to pull these
+images over the network from a container registry.
+
+A container image is embedded by adding a reference to it to the Image
+Builder blueprint like this:
+
+[source,terminal]
+----
+[[containers]]
+source = "<image_pullspec_with_tag_or_digest>"
+----
+
+To embed MicroShift's container images, you need to know the exact list of
+container image references used by the MicroShift version you are embedding.
+You can obtain this list from the `microshift-release-info` RPM package
+matching the version of the `microshift` RPM in your blueprint using the
+following procedure.
+
+.Procedure
+
+. Download the `microshift-release-info` RPM package matching your MicroShift version using the following command:
++
+[source,terminal]
+----
+$ dnf download microshift-release-info-{ocp-version}.0-1.el{rhel-major}
+----
+
+. Exctract the release info by entering the following command:
++
+[source,terminal]
+----
+$ rpm2cpio microshift-release-info-{ocp-version}.0-1.el{rhel-major}.noarch.rpm | cpio -idmv
+./usr/share/microshift/release/release-aarch64.json
+./usr/share/microshift/release/release-x86_64.json
+----
+
+. Using the release info for your CPU architecture, you can now generate the lines to append to your blueprint by entering the following command:
++
+[source,terminal]
+----
+$ jq -r '.images | .[] | ("[[containers]]\nsource = \"" + . + "\"\n")' ./usr/share/microshift/release/release-$(uname -i).json
+[[containers]]
+source = "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9945c3f5475a37e145160d2fe6bb21948f1024a856827bc9e7d5bc882f44a750"
+
+[[containers]]
+source = "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:82cfef91557f9a70cff5a90accba45841a37524e9b93f98a97b20f6b2b69e5db"
+
+...
+----
+
+== Provisioning with support for MicroShift
+The machine you are provisioning must meet the xref:../microshift_install/microshift-install-rpm.adoc#system-requirements-installing-microshift[system requirements for installing MicroShift].
+
+When provisioning a machine using your {op-system-ostree} image following any of the
+methods described in the https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/composing_installing_and_managing_rhel_for_edge_images/index[{op-system-ostree} documentation], you must configure the system to meet the
+following prerequisites:
+
+* You must provide a logical volume manager (LVM) volume group (VG) with sufficient capacity for the persistent volumes (PVs) of your workload.
```

Comment:
Do we have procedures for the first two items?
Can xref our firewall procedure

Reply:
LVM is a standard RHEL thing, I don't think we want to redocument it here. If we can easily link to some RHEL doc, that's fine, but I think we have to expect a certain level of understanding of operating system concepts in these docs so if there's no easy link available I think we can get along fine without it.